### PR TITLE
[1798] mcb providers create

### PR DESCRIPTION
**Pull request by richrace**
### Context

Need to provide a command via `mcb` to create a Provider (and organisation).

### Changes proposed in this pull request

- Add create command for Providers 
- Automatically set `accrediting_provider` and `scitt` based on Provider Type
- `User.admins` scope

### Guidance to review

- ~~Should this belong here: https://github.com/DFE-Digital/manage-courses-backend/pull/683/files#diff-5c08de53fa6bea147e31fe1d5930422aR30-R32~~

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by richrace** [comment]
[mcb providers create](https://trello.com/c/dXGRMUmo/1798-mcb-providers-create)

**Comment by benilovj** [comment]
Looking good! I'd still be tempted to move 

```
provider.scheme_member = 'Y'
provider.year_code = provider.recruitment_cycle.year
...
provider.changed_at = Time.now.utc
```

into the `Provider` model but I don't think it's a deal-breaker for me.

File: app/models/recruitment_cycle.rb [hunk 1]
```diff
@@ -28,6 +28,10 @@ def self.next_recruitment_cycle
     all.order(:year).second
   end
 
+  def next
+    RecruitmentCycle.find_by(year: year.to_i + 1)
+  end
```
Comment (misaka) [inline app/models/recruitment_cycle.rb]: Even though this is the right implementation, I feel like this muddies the waters a bit ... wrt `RecruitmentCycle#next_recruitment_cycle`. I know this is a pretty big PR already, but is there anything simple we can do to not have this split-view of what qualifies as the "next" cycle?

File: lib/mcb/provider_editor.rb [hunk 2]
```diff
@@ -69,7 +171,8 @@ def update(attrs)
     end
 
     def check_authorisation
-      raise Pundit::NotAuthorizedError unless ProviderPolicy.new(@requester, @provider).update?
+      action = @provider.persisted? ? :update? : :create?
+      raise Pundit::NotAuthorizedError unless ProviderPolicy.new(@requester, @provider).send(action)
```
Comment (misaka) [inline lib/mcb/provider_editor.rb]: Minor improvement, but this use's Pundit's built-in mechanism to find `ProviderPolicy`.

```suggestion
      raise Pundit::NotAuthorizedError unless Pundit.policy(@requester, @provider).send(action)
```

File: app/models/recruitment_cycle.rb (hunk 1 above)
Comment (benilovj) [reply app/models/recruitment_cycle.rb]: Would it help if we changed:

```ruby
  def self.next_recruitment_cycle
    current_recruitment_cycle.next
  end
```

Comment (misaka) [reply app/models/recruitment_cycle.rb]: Ya, I thought that would work too. 👍

File: lib/mcb/provider_editor.rb (hunk 2 above)
Comment (benilovj) [reply lib/mcb/provider_editor.rb]: fixed and force-pushed.

File: app/models/recruitment_cycle.rb (hunk 1 above)
Comment (benilovj) [reply app/models/recruitment_cycle.rb]: Right-o, fixed in 4d25757

**Pull request merged**